test/ValidationFramework/Parser/Parser.hpp

```diff
@@ -0,0 +1,149 @@
+/// Apache License 2.0
```

Comment:
So, the `parser` is a class holding a bunch of methods that allow to convert a yaml (and maybe in the future other input formats if needed, like `xml` or else as this is usually used in CDM, OEM and others as per CCSDS recommendations) into OSTk objects (`SatelliteSystem`, `NumericalSolver`, `Propagator`), IMO it can be used outside of the testing and validation framework, it is another way to configure OSTk, do you agree?

Reply:
The apps are different as they need to match other apps from other tools etc, but still, there might be some benefits in having them outside of `tests` or `validation` IMO

Reply:
Then we should use those concepts in `tests` and `validation` ofc to standardize the way we set them up and perform comparison but it's one use case for `Parser` etc, do you agree?

Reply:
Okay perfect aligned. Let's keep them there for now I agree ✅